### PR TITLE
Makes dictionary instead of string for visual shader version

### PR DESCRIPTION
**Pull request by Chaosus**
Replaced a String with Dictionary with ["major"] and ["minor"] members, I think it's better for further usage and parsing + added a string to `update_shader` printed for every auto-changed shader like:

`The shader ("res://test.tres") has been updated to correspond Godot 4.0 version.`

### CODE REVIEW

File: doc/classes/VisualShader.xml [hunk 1]
```diff
@@ -149,7 +149,7 @@
 		<member name="graph_offset" type="Vector2" setter="set_graph_offset" getter="get_graph_offset" default="Vector2(0, 0)">
 			The offset vector of the whole graph.
 		</member>
-		<member name="version" type="String" setter="set_version" getter="get_version" default="&quot;&quot;">
+		<member name="version" type="Dictionary" setter="set_version" getter="get_version" default="{}">
 		</member>
```
Comment (Calinou) [inline doc/classes/VisualShader.xml:153]:
```suggestion
		<member name="version" type="Dictionary" setter="set_version" getter="get_version" default="{}">
			The Godot version this [VisualShader] was designed for, in the form of a [Dictionary]. Example: [code]{"major": 4, "minor": 0}[/code]
			This is used by the editor to convert visual shaders from older Godot versions.
		</member>
```

**Review by Calinou — CHANGES_REQUESTED** [review]

File: doc/classes/VisualShader.xml (hunk 1 above)
Comment (Chaosus) [reply doc/classes/VisualShader.xml]: @Calinou Thanks, I've updated it and make you a co-author.

**Review by Calinou — APPROVED** [review]
Documentation looks good to me. That said, maybe the property should be renamed `engine_version`? I'm not sure.

**Comment by Chaosus** [comment]
Yeah, I think it's a good idea too - (rename to `engine_version`).

**Comment by Chaosus** [comment]
Hmm, I've detected that there is no way to detect that a VisualShader has been created in the version before 4.0 since the was no property before it, so _update_engine_version has been called for newly created shaders too because they are initialized with empty `Dictionary`, because of that I replaced warning printing for versions after 4.0 (a future).

**Pull request merged**